java/custom/com/sun/cldc/i18n/j2me/UTF_8_Reader.java

```diff
@@ -56,28 +98,173 @@ public Reader open(InputStream in, String enc)
      * @exception IOException is thrown if the input stream 
      * could not be read for the raw unconverted character
      */
-    public native int readNative(char cbuf[], int off, int len);
     public int read(char cbuf[], int off, int len) throws IOException {
-        if (!initialized) {
```

Comment:
I think we should keep this optimization, otherwise we'd end up reading a few bytes at a time from the InputStream. In the case of a FCInputStream, we'd call the FS native a lot of times.